### DISCUSSION TITLE
[QUESTION] IDEA StatiCa RCS: print PDF reports with API

### DISCUSSION THREAD

**Discussion by RayaanABT** [idea-statica/ideastatica-public#148] [Features and Ideas]
Is it possible to print PDF reports with an API command?

**Comment by mpospisil** [comment]
It is not possible now.

**Reply by Balkmannen** [reply]
Is there any plan to implement it soon?

**Reply by Kar-Myl** [reply]
@mpospisil is there any update?

**Reply by nathanlukeIDEA** [reply]
Hi @Balkmannen and @Kar-Myl, yes this is now available as of 23.1.4. :) https://developer.ideastatica.com/api-plugin/IdeaStatiCa.Plugin.ConnectionHiddenCheckClient.html#IdeaStatiCa_Plugin_ConnectionHiddenCheckClient_GenerateReportPdf_System_String_System_String_IdeaStatiCa_Plugin_ConnReportSettings_

**Reply by khameeteman** [reply]
Hi @nathanlukeIDEA I believe you are referring to "Connection" right? When will it be available for "RCS"?

**Reply by nathanlukeIDEA** [reply]
HI @khameeteman, sorry I do not have a timeframe on when it will be available for RCS API.

**Comment by Kar-Myl** [comment]
> Hi @Balkmannen and @Kar-Myl, yes this is now available as of 23.1.4. :) https://developer.ideastatica.com/api-plugin/IdeaStatiCa.Plugin.ConnectionHiddenCheckClient.html#IdeaStatiCa_Plugin_ConnectionHiddenCheckClient_GenerateReportPdf_System_String_System_String_IdeaStatiCa_Plugin_ConnReportSettings_

Hi @nathanlukeIDEA , could you please provide a python example for report pdf generation? I already have a .ideaCon file that I would like to load and then generate the report for. The three arguments required ( binding: Binding, remoteAddress: EndpointAddress, logger: IPluginLogger ) are not clear to us. 

Thanks in advance!

**Reply by nathanlukeIDEA** [reply]
Hi @Kar-Myl,

thanks for your patience. Please find below a python example.


```python
import clr
import sys
import json
import math
import os

idea_path = "C:\Program Files\IDEA StatiCa\StatiCa 23.1"
sys.path.append(idea_path);

clr.AddReference('IdeaStatiCa.Plugin')

import IdeaStatiCa.Plugin
from IdeaStatiCa.Plugin import ConnHiddenClientFactory

factory = ConnHiddenClientFactory(idea_path)
client = factory.Create()

#path to Idea file
client.OpenProject('test.ideaCon')

#get first connection in project
projectInfo = client.GetProjectInfo()
connections = projectInfo.Connections
connection = connections[0].Identifier

#calculate project
result = client.Calculate(connection)

#print report
report_path = "test_output.pdf"
settings = IdeaStatiCa.Plugin.ConnReportSettings()
client.GenerateReportPdf(connection, report_path, settings)

#close project and client
client.CloseProject()
client.Close()
```

**Reply by Kar-Myl** [reply]
Hi @nathanlukeIDEA ,

This solution works and is quite useful. Thank you so much!

Could you please provide some information about the ConnReportSettings class? I checked the documentation here (https://developer.ideastatica.com/api-plugin/IdeaStatiCa.Plugin.ConnReportSettings.html?q=ConnReportSettings) but couldn't find any of the methods in it.

I am interested in knowing what could be done in order to generate a detailed report with the theoretical background or the other options available in the software.

**Reply by nathanlukeIDEA** [reply]
Hi @Kar-Myl, We cannot set the avaliable settings yet unfortunately so the empty settings in just a placeholder at the moment.

**Reply by Kar-Myl** [reply]
Okay, thank you for letting me know! 😄

**Reply by gerso-magro** [reply]
Have you already worked on a way to change the settings within a report?

**Reply by nathanlukeIDEA** [reply]
Hi, it is still not possible, although we are looking into adding it. You can follow this issue. I will post here when it is provided. https://github.com/idea-statica/ideastatica-public/issues/495

**Reply by karol153** [reply]
Hi, is there any progres on adding report settings?

**Reply by nathanlukeIDEA** [reply]
@karol153, we are making progress on this. As I mentioned above, I will provide information on this issue when it is complete. We aim on a timeframe of release for v25.1. https://github.com/idea-statica/ideastatica-public/issues/495

**Reply by nathanlukeIDEA** [reply]
Hi @gerso-magro, @karol153, @Kar-Myl. In v25.1, we have implemented a new settings controller where you can interact with the project settings, including the report settings. https://developer.ideastatica.com/docs/api/connection-api/clients/csharp/docs/SettingsApi.html#getsettings
